nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -598,23 +598,39 @@ sub multiLineList {
 ${\join "", (map { "  $_\n" } (uniq @attrs))}}
 EOF
 
-sub generateNetworkingDhcpConfig {
-    # FIXME disable networking.useDHCP by default when switching to networkd.
+sub generateNetworkingConfig {
     my $config = <<EOF;
-  # Enables DHCP on each ethernet and wireless interface. In case of scripted networking
-  # (the default) this is the recommended approach. When using systemd-networkd it's
-  # still possible to use this option, but it's recommended to use it in conjunction
-  # with explicit per-interface declarations with `networking.interfaces.<interface>.useDHCP`.
-  networking.useDHCP = lib.mkDefault true;
+  # Use systemd-networkd to enable DHCP (IPv4) and Router
+  # Advertisements (IPv6) on every ethernet and wireless interface.
+  systemd.network = {
+    enable = true;
+    networks = {
+      # Use lower numbers for additional profiles, since networkd
+      # processes these alphanumerically and every interface will only
+      # ever use its first match.
+      "99-autoconfig-ether" = {
+        matchConfig.Type = [
+          "ether"
```

Comment:
I think we also want `Kind = !*` here.

See https://github.com/systemd/systemd/commit/e5188e4f9b814b89f5c89624c0762e3ecba7bd11